src/main/scala/io/joern/typestubs/Main.scala

```diff
@@ -1,11 +1,71 @@
 package io.joern.typestubs
 
 import io.joern.typestubs.ruby.BuiltinPackageDownloader
+import scopt.OParser
+
+final case class Config(writeToJson: Boolean = false, languageFrontend: String = "all") {
+
+  def writeToJson(value: Boolean): Config = {
+    copy(writeToJson = value)
+  }
+
+  def withLanguageFrontend(value: String): Config = {
+    copy(languageFrontend = value)
+  }
+}
+
+private object Frontend {
+
+  implicit val defaultConfig: Config = Config()
+
+  val cmdLineParser: OParser[Unit, Config] = {
+    val availableFrontendLanguages = Seq("", "ruby", "chsarp", "all")
+
+    val builder = OParser.builder[Config]
+    import builder.*
+    OParser.sequence(
+      programName("joern-type-stubs"),
+      opt[Unit]("writeToJson")
+        .action((_, c) => c.writeToJson(true))
+        .text("Write builtin types to JSON files without zipping, instead of MessagePack files with a zipped folder"),
+      opt[String]("withLanguageFrontend")
+        .action((x, c) => c.withLanguageFrontend(x))
+        .validate {
+          case x if availableFrontendLanguages.contains(x) => success
+          case x if !availableFrontendLanguages.contains(x) =>
+            failure(s"Only available languages are: [${availableFrontendLanguages.mkString(", ")}]")
+        }
+        .text("""The Frontend Language to generate builtin types for, defaults to `all`:
+            |ruby -> Ruby
```

Comment:
We could make this simpler
```suggestion
            |The Frontend Language to generate builtin types for, defaults to `all`: [${availableFrontendLanguages.mkString(","}]
```